keanu-project/src/test/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensorTest.java

```diff
@@ -638,4 +641,20 @@ public void youCanFixACustomValidationIssueByReplacingTheValue() {
         containsMinusOne = validator.validate(containsMinusOne);
         assertThat(containsMinusOne, equalTo(expectedResult));
     }
+
+    @Test
+    public void comparesIntegerTensorWithScalar() {
+        IntegerTensor value = IntegerTensor.create(1, 2, 3);
+        IntegerTensor differentValue = IntegerTensor.create(1);
+        BoolVertex result = new EqualsVertex<>(ConstantVertex.of(value), ConstantVertex.of(differentValue));
```

Comment:
If we are using `EqualsVertex`, shouldn't this be in `EqualsVertexTest`? Otherwise just test `elementWiseEquals` directly.

Reply:
No - we're testing tensor logic here. But yeah - I've removed the unnecessary creation of the vertices and kept them as tensors